src/components/splitButton/splitButton.ts

```diff
@@ -0,0 +1,400 @@
+  @property({ type: Boolean, reflect: true })
+  destructive = false;
+
+  /** Menu CSS min-width value. */
+  @property({ type: String })
+  menuMinWidth = 'initial';
+
+  /** Listbox/menu open state. */
+  @property({ type: Boolean })
+  open = false;
+
+  /**
+   * Queries any slotted options.
+   * @ignore
+   */
+  @queryAssignedElements({ selector: 'kd-splitbutton-option' })
+  options!: Array<any>;
+
+  /**
+   * Open menu upwards.
+   * @ignore
+   */
+  @state()
+  _openUpwards = false;
+
+  /**
+   * Queries the .select DOM element.
+   * @ignore
+   */
+  @query('.select')
+  buttonEl!: HTMLElement;
+
+  /**
+   * Assistive text for screen readers.
+   * @ignore
+   */
+  @state()
+  assistiveText = 'Split button menu options.';
+
+  override render() {
+    const typeClassMap = {
+      [SPLIT_BTN_KINDS.PRIMARY_APP]: 'primary-app',
+      [SPLIT_BTN_KINDS.PRIMARY_WEB]: 'primary-web',
+      [SPLIT_BTN_KINDS.SECONDARY]: 'secondary',
+    };
+
+    const baseTypeClass = typeClassMap[this.kind];
+    const destructModifier = this.destructive ? '-destructive' : '';
+
+    const classes = {
+      button: true,
+      [`kd-split-btn--${baseTypeClass}${destructModifier}`]: true,
+      [`kd-split-btn--${baseTypeClass}`]: !this.destructive,
+      'kd-split-btn--large': this.size === SPLIT_BTN_SIZES.LARGE,
+      'kd-split-btn--small': this.size === SPLIT_BTN_SIZES.SMALL,
+      'kd-split-btn--medium': this.size === SPLIT_BTN_SIZES.MEDIUM,
+      [`kd-split-btn--icon-${this.iconPosition}`]: !!this.iconPosition,
+    };
+
+    return html`
+      <div class="split-btn-wrapper ${this.open ? 'open' : ''}">
+        <!-- label button UI -->
+        <button
+          class=${classMap({ ...classes, 'kd-split-btn-label': true })}
+          type="button"
+          ?disabled=${this.disabled}
+          aria-label=${ifDefined(this.description)}
+          title=${ifDefined(this.description)}
+          name=${ifDefined(this.name)}
+        >
+          <span>
+            ${this.label}
+            <slot name="icon"></slot>
+          </span>
+        </button>
+        <!-- menu button UI -->
+        <button
+          class=${classMap({
+            ...classes,
+            select: true,
+            [`kd-split-btn--${this.size}-arrow-btn`]: true,
+            'kd-split-btn-icon': true,
+            'kd-split-btn-margin-overlapped':
+              this.kind === SPLIT_BTN_KINDS.SECONDARY,
+          })}
+          type="button"
+          ?disabled=${this.disabled}
+          aria-label=${ifDefined(this.description)}
+          title=${ifDefined(this.description)}
+          aria-haspopup="true"
+          aria-expanded="false"
+          @click=${this.toggleDropdown}
+          @keydown=${(e: any) => this.handleButtonKeydown(e)}
+        >
+          <span>
+            <kd-icon class="arrow-icon" .icon=${downIcon}></kd-icon>
+          </span>
+        </button>
+        <!-- Split Button Menu UI -->
+        <ul
```

Comment:
I get an accessibility violation on this element when I run the test with the dropdown open.

Reply:
I didn't get any at my end. Can you share what's the violation ?

Reply:
found it actually **id="label-{this.name}"** was missing. So putting this eliminates the violation